src/react/hooks/use-occupancy.ts

```diff
@@ -60,16 +61,20 @@ export const useOccupancy = (params?: UseOccupancyParams): UseOccupancyResponse
     presenceMembers: 0,
   });
 
+  // create stable references for the listeners
+  const listenerRef = useEventListenerRef(params?.listener);
+  const onDiscontinuityRef = useEventListenerRef(params?.onDiscontinuity);
+
   // if provided, subscribes the user provided discontinuity listener
   useEffect(() => {
-    if (!params?.onDiscontinuity) return;
+    if (!onDiscontinuityRef) return;
     logger.debug('useOccupancy(); applying onDiscontinuity listener', { roomId: room.roomId });
-    const { off } = room.occupancy.onDiscontinuity(params.onDiscontinuity);
+    const { off } = room.occupancy.onDiscontinuity(onDiscontinuityRef);
```

Comment:
Is this supposed to be onDiscontinuityRef.current?

Reply:
No - because under the hood its just a function, not the return type of `React.useRef`